### PR TITLE
TruncateText for challenge description over 50 characters

### PR DESCRIPTION
**Pull request by El-Duckster**
This PR introduces a utility function (truncateText) to truncate long descriptions at word boundaries and integrates it into the Challenge class to enhance the display of descriptions.

**Details of the Solution:**
1- Helper Function:
a- Added truncateText in utilities/truncateText.js.
b- Truncates text at word boundaries, appending "..." if the text exceeds the specified length.

2- Integration:
a- Updated the Challenge class to use truncateText when rendering descriptions.
b- Applied truncation with a max length of 50 characters.

3-Refactoring: Centralized truncation logic into a reusable utility for consistency.
tested on multiple screen sizes to ensure layout stability. abd browsers such as google chrome, och Mozilla Firefox

Solves issue #101

### CODE REVIEW

File: src/js/utilities/truncateText.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    if (text.length <= maxLength) return text;
+  
+    const truncated = text.slice(0, maxLength).trimEnd();
+    const lastSpaceIndex = truncated.lastIndexOf(' ');
+  
+    return lastSpaceIndex > 0
+      ? truncated.slice(0, lastSpaceIndex) + '...'
+      : truncated + '...';
+  };
+
```
Comment (Kirill9m) [inline src/js/utilities/truncateText.js:12]: Your truncateText function is really good and functional! 
A very clean and correct code!
It is great that you are using functions like slice, trimEnd() and lastIndexOf()

**Review by Kirill9m — APPROVED** [review]
I fully approve this pr!

**Review by Deansie — APPROVED** [review]
I have nothing to add, looks good. Well done!

**Comment by El-Duckster** [comment]
[Begränsa beskrivningstext till 50 tecken i kort](https://trello.com/c/fordIxi9/28-begr%C3%A4nsa-beskrivningstext-till-50-tecken-i-kort)

**Pull request merged**